### PR TITLE
Iterating over empty arrays

### LINKED ISSUE #259: iterating over empty arrays problem
I ran over this usecase where a loop is supposed to iterate over an empty array and I get an unclear message

Command failed java.lang.RuntimeException: Slang Error : Error running: 'get_parent_image'

    null

As @orius123 we should either show a warning about not being allowed to iterate over empty arrays or just skip the loop

### PR DESCRIPTION
**Pull request by Bonczidai**
fixes #259

### CODE REVIEW

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/AsyncLoopBinding.java [hunk 1]
```diff
@@ -30,21 +30,26 @@
     @Autowired
     private ScriptEvaluator scriptEvaluator;
 
+    public static String generateAsyncLoopExpressionMessage(String nodeName, String message) {
+        return "Error evaluating async loop expression in task '" + nodeName + "', error is: \n" + message;
```
Comment (meirwah) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/AsyncLoopBinding.java]: take static strings out, to be private const ststic...

File: cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/LoopsBinding.java [hunk 2]
```diff
@@ -107,6 +107,9 @@ private LoopCondition createForLoopCondition(LoopStatement forLoopStatement, Con
                     "in task: '" + nodeName + "' " +
                     "doesn't return an iterable, other types are not supported");
         }
+        if (!forLoopCondition.hasMore()) {
+            throw new RuntimeException("Error evaluating for loop expression in task '" + nodeName + "',\n\tError is: expression is empty");
```
Comment (meirwah) [inline cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/bindings/LoopsBinding.java]: take strings out...

**Comment by orius123** [comment]
what meir said and merge :100:

**Comment by Bonczidai** [comment]
also added tests

**Comment by meirwah** [comment]
:+1:

**Pull request merged**